cartesian_trajectory_interpolation/src/cartesian_trajectory.cpp

```diff
@@ -55,8 +56,31 @@ bool CartesianTrajectory::init(const cartesian_control_msgs::CartesianTrajectory
     {
       return false;
     }
-    CartesianTrajectorySegment s(i->time_from_start.toSec(), CartesianState(*i), std::next(i)->time_from_start.toSec(),
-                                 CartesianState(*std::next(i)));
+
+    CartesianState state = CartesianState(*i);
+    CartesianState next_state = CartesianState(*std::next(i));
+
+    if (sign_flipped)
+    {
+      state.q.x() = state.q.x() * -1;
+      state.q.y() = state.q.y() * -1;
+      state.q.z() = state.q.z() * -1;
+      state.q.w() = state.q.w() * -1;
```

Comment:
```suggestion
      state.q = state.q.conjugate();
```

Reply:
That won't negate the *real* part of the quaternion, but I think it's cleaner. Does that still work?